src/Compilers/CSharp/Portable/FlowAnalysis/EmptyStructTypeCache.cs

```diff
@@ -166,6 +166,14 @@ private FieldSymbol GetActualInstanceField(Symbol member, NamedTypeSymbol type)
                 switch (member.Kind)
                 {
                     case SymbolKind.Field:
+                        // Do not report virtual tuple fields.
+                        // They are additional aliases to the fields of the underlying struct or nested extensions.
+                        // and as such are already accounted for via the nonvirtual fields.
+                        if (field is TupleVirtualElementFieldSymbol)
```

Comment:
Can you please make this the test a new internal virtual boolean property (`FieldSymbol.IsVirtualTupleElement`) rather than testing for particular implementation types? The latter makes the code more fragile as we refactor inheritance hierarchies, while the virtual property would make the dependency explicit.

---
In reply to: [74524647](https://github.com/dotnet/roslyn/pull/13107#discussion_r74524647) [](ancestors = 74524647)